diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatGradlePlugin.kt

```diff
@@ -0,0 +1,52 @@
+package org.cqfn.diktat.plugin.gradle
+
+import com.pinterest.ktlint.reporter.plain.PlainReporter
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.artifacts.ExternalModuleDependency
+
+/**
+ * Plugin that configures diktat and registers tasks to run diktat
+ */
+@Suppress("unused", "MagicNumber")
+class DiktatGradlePlugin : Plugin<Project> {
+    /**
+     * @param project a gradle [Project] that the plugin is applied to
+     */
+    override fun apply(project: Project) {
+        val diktatExtension = project.extensions.create(DIKTAT_EXTENSION, DiktatExtension::class.java)
+        diktatExtension.inputs = project.fileTree("src").apply {
+            include("**/*.kt")
+        }
+        diktatExtension.reporter = PlainReporter(System.out)
+
+        val gradleVersion = GradleVersion.fromString(project.gradle.gradleVersion)
+
+        // only gradle 7+ (or maybe 6.8) will embed kotlin 1.4+, kx.serialization is incompatible with kotlin 1.3, so until then we have to use JavaExec wrapper
+        // FixMe: when gradle with kotlin 1.4 is out, proper configurable tasks should be added
+        if (/*gradleVersion.major < 6 || gradleVersion.major == 6 && gradleVersion.minor < 8*/ true) {
+            // configuration to provide JavaExec with correct classpath
+            val diktatConfiguration = project.configurations.create(DIKTAT_CONFIGURATION) { configuration ->
+                configuration.isVisible = false
+                configuration.dependencies.add(project.dependencies.create("org.jetbrains.kotlin:kotlin-stdlib:1.4.10"))
```

Comment:
is there any way not to hardcode dependencies?

Reply:
I've added passing versions via properties from maven and a little of codegen to use them in plugin.